sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java

```diff
@@ -3539,6 +3539,7 @@ public void onTimer() {}
     @Category({
       ValidatesRunner.class,
       UsesTimersInParDo.class,
+      UsesUnboundedPCollections.class,
```

Comment:
It might be better to follow-up with the PR author as to why they added `.setIsBoundedInternal(IsBounded.UNBOUNDED)` instead of leaving it be bounded and have different variants of runners execute the test (batch/streaming/...)

Reply:
I think it was so that it would run on a streaming runner (especially the Dataflow runner). ValidatesRunner tests that use TestStream don't run on Dataflow today I believe, and we have several other places in our test codebase where this pattern is used.

Reply:
I split the test into batch and streaming variants, PTAL

Reply:
ping @lukecwik @reuvenlax